Neos.Flow/Tests/Unit/Mvc/Controller/ActionControllerTest.php

```diff
@@ -246,6 +250,33 @@ public function processRequestInjectsSettingsToView()
         $this->actionController->processRequest($this->mockRequest, $mockResponse);
     }
 
+    /**
+     * @test
+     */
+    public function processRequestSetsNegotiatedContentTypeOnResponse()
+    {
+        $this->actionController = $this->getAccessibleMock(ActionController::class, ['resolveActionMethodName', 'initializeActionMethodArguments', 'initializeActionMethodValidators', 'resolveView', 'callActionMethod', 'initializeController']);
+
+        $this->inject($this->actionController, 'objectManager', $this->mockObjectManager);
+        $this->inject($this->actionController, 'controllerContext', $this->mockControllerContext);
+        $this->inject($this->actionController, 'request', $this->mockRequest);
+
+        $this->inject($this->actionController, 'arguments', new Arguments([]));
+
+        $mockMvcPropertyMappingConfigurationService = $this->createMock(Mvc\Controller\MvcPropertyMappingConfigurationService::class);
+        $this->inject($this->actionController, 'mvcPropertyMappingConfigurationService', $mockMvcPropertyMappingConfigurationService);
+
+        $mockHttpRequest = $this->getMockBuilder(ServerRequestInterface::class)->disableOriginalConstructor()->getMock();
+        $this->mockRequest->expects(self::any())->method('getHttpRequest')->willReturn($mockHttpRequest);
+
+        $mockResponse = new Mvc\ActionResponse;
+        $this->inject($this->actionController, 'response', $mockResponse);
+        $this->inject($this->actionController, 'negotiatedMediaType', 'application/json');
```

Comment:
Meh, this is a really bad test... it should inject supportedMediaTypes into the controller and not test this implementation detail